activestorage/test/test_helper.rb

```diff
@@ -60,6 +60,23 @@ class ActiveSupport::TestCase
     ActiveStorage::Current.reset
   end
 
+  def assert_queries(expected_count)
+    ActiveRecord::Base.connection.materialize_transactions
+
+    queries = []
+    ActiveSupport::Notifications.subscribe("sql.active_record") do |*, payload|
+      queries << payload[:sql] unless %w[ SCHEMA TRANSACTION ].include?(payload[:name])
+    end
+
+    yield.tap do
+      assert_equal expected_count, queries.size, "#{queries.size} instead of #{expected_count} queries were executed. #{queries.inspect}"
+    end
+  end
+
+  def assert_no_queries(&block)
+    assert_queries(0, &block)
+  end
```

Comment:
This could probably do with extracting along with the copies in Action Text and Action View into a module - not for this PR though.